lifecycle/CMakeLists.txt

```diff
@@ -11,55 +11,28 @@ if(CMAKE_COMPILER_IS_GNUCXX OR CMAKE_CXX_COMPILER_ID MATCHES "Clang")
   add_compile_options(-Wall -Wextra -Wpedantic)
 endif()
 
-find_package(ament_cmake REQUIRED)
-find_package(rclcpp_lifecycle REQUIRED)
-find_package(lifecycle_msgs REQUIRED)
-find_package(rclcpp REQUIRED)
-find_package(std_msgs REQUIRED)
+find_package(ament_cmake_auto REQUIRED)
+ament_auto_find_build_dependencies()
```

Comment:
We don't really use this feature in the core ROS 2 packages, though I don't know the reason why.
Since we seem to be moving closer to vanilla CMake packages, I would rather see the `ament_cmake_auto` addition proposed as a separate PR to discuss separately from the refactor of the lifecycle demos code.

Reply:
Yes, agreed.  The main reason we don't use it is exactly to move towards more vanilla CMake, so I would suggest not doing this and simplifying this patch.